### ISSUE TITLE
Allow running `pnpm install` from lockfile without the "is lockfile out-of-sync" check

### ISSUE THREAD

**Issue by pkerschbaum** [pnpm/pnpm#7236]
### Contribution

- [X] I'd be willing to implement this feature ([contributing guide](https://github.com/pnpm/pnpm/blob/main/CONTRIBUTING.md))

### Describe the user story

At @hokify we have a `.pnpmfile.cjs` which adjusts some dependencies.  
During build in our CI/CD system we would like to install dependencies from the lockfile without the logic of `.pnpmfile.cjs` being run. For the reason that we try to utilize the Docker build cache as much as possible, and less files being needed means higher chance to hit the cache.

Currently we run `pnpm install --frozen-lockfile` which fails if `.pnpmfile.cjs` is missing (the dependency graph does not match with the lockfile then).

### Describe the solution you'd like

It would be great if there was an option to just assume the lockfile is up-to-date and skips the "out-of-sync" check. Something like `pnpm install --from-lockfile`.  
Actually, `pnpm fetch --frozen-lockfile` does that already, it does not need `.pnpmfile.cjs`.

### Describe the drawbacks of your solution

We have to copy `.pnpmfile.cjs` into Docker builds to get `pnpm install --frozen-lockfile` runnable.

### Describe alternatives you've considered

I tried `pnpm install --frozen-lockfile --offline` but that fails with the same error.

**Comment by KSXGitHub** [comment]
Why use `.pnpmfile.cjs` over overrides? Is your adjustment too complex to express it declaratively?

**Comment by KSXGitHub** [comment]
> We have some workspace projects which depend on other workspace projects, and during build, the projects _transpile their upstream workspace dependencies_.

Looks like a private registry would fit your use case better. You can also use private git repos (with SSH keys).

Anyway, can your `.pnpmfile.cjs` detect the environment it is running on (by checking `CI` environment variable for example)? If not, can your build scripts detect the cache? Using pnpmfile in place of build scripts is some hacky solution.

**Comment by pkerschbaum** [comment]
I agree that there are better ways to handle this, e.g. ensuring via some CI automation that the package.json of an application includes the dependencies, devDependencies of any workspace project which it transpiles during build.  
Therefore, closing this issue.  
Thx for your suggestions!
